idaes/models_extra/gas_solid_contactors/unit_models/tests/test_FB1D.py

```diff
@@ -0,0 +1,1367 @@
+        )
+
+        def gas_holdup(b, x):
+            tf = b.flowsheet().time.last()
+            t0 = b.flowsheet().time.first()
+            calculate_variable_from_constraint(
+                b.gas_phase.properties[tf, x].mw,
+                b.gas_phase.properties[tf, x].mw_eqn,
+            )
+            calculate_variable_from_constraint(
+                b.gas_phase.properties[t0, x].mw,
+                b.gas_phase.properties[t0, x].mw_eqn,
+            )
+            mass_final_per_length = (
+                b.gas_phase.properties[tf, x].dens_mol
+                * b.gas_phase.properties[tf, 1].mw
+                * b.gas_phase.area[tf, x]
+            )
+            mass_initial_per_length = (
+                b.gas_phase.properties[t0, x].dens_mol
+                * b.gas_phase.properties[t0, 1].mw
+                * b.gas_phase.area[t0, x]
+            )
+            return (mass_final_per_length - mass_initial_per_length) * b.bed_height
+
+        iron_oc.fs.unit.gas_holdup = Integral(
+            iron_oc.fs.unit.length_domain,
+            wrt=iron_oc.fs.unit.length_domain,
+            rule=gas_holdup,
+            doc="Gas mass holdup, kg",
+        )
+
+        def oc_at_initial_time(b, x):
+            t0 = b.flowsheet().time.first()
+            calculate_variable_from_constraint(
+                b.solid_properties[t0, x].dens_mass_particle,
+                b.solid_properties[t0, x].density_particle_constraint,
+            )
+            mass_per_length = (
+                b.solid_properties[t0, x].dens_mass_particle * b.solid_phase_area[t0, x]
+            )
+            return mass_per_length * b.bed_height
+
+        iron_oc.fs.unit.oc_at_initial_time = Integral(
+            iron_oc.fs.unit.length_domain,
+            wrt=iron_oc.fs.unit.length_domain,
+            rule=oc_at_initial_time,
+            doc="OC at initial reaction time, kg",
+        )
+
+        def oc_at_final_time(b, x):
+            tf = b.flowsheet().time.last()
+            calculate_variable_from_constraint(
+                b.solid_properties[tf, x].dens_mass_particle,
+                b.solid_properties[tf, x].density_particle_constraint,
+            )
+            mass_per_length = (
+                b.solid_properties[tf, x].dens_mass_particle * b.solid_phase_area[tf, x]
+            )
+            return mass_per_length * b.bed_height
+
+        iron_oc.fs.unit.oc_at_final_time = Integral(
+            iron_oc.fs.unit.length_domain,
+            wrt=iron_oc.fs.unit.length_domain,
+            rule=oc_at_final_time,
+            doc="OC at final reaction time, kg",
+        )
+
+        mass_change_gas = value(
+            iron_oc.fs.unit.gas_input_total
+            - iron_oc.fs.unit.gas_output_total
+            - iron_oc.fs.unit.gas_holdup
+        )
+        mass_change_solid = value(
+            iron_oc.fs.unit.oc_at_initial_time - iron_oc.fs.unit.oc_at_final_time
+        )
+
+        assert mass_change_gas <= 1e-8
```

Comment:
The tests for the 0D fixed bed assert against the absolute values of the changes, presumably to avoid issues with negative differences. Is there a chance this could occur with the test problem formulation? If so, I would replace line 825 with `assert abs(mass_change_gas) <= 1e-8)`, and similarly on line 826.

Reply:
Fixed.